### ISSUE TITLE
Problems when using function “bitr_kegg”

### ISSUE THREAD

**Issue by chenziru** [YuLab-SMU/clusterProfiler#219]
The KEGG gene ID and the ncbi-geneID should be exactly the same according to [KEGG web site](http://rest.kegg.jp/conv/ncbi-geneid/hsa) .

However,  ID conversion from 'ncbi-geneid' to 'kegg' and ID conversion from 'kegg'  to 'ncbi-geneid' using function “bitr_kegg”,  I get different results: 
- conversion from 'ncbi-geneid' to 'kegg':  100% of input gene IDs are fail to map...
- conversion from 'kegg'  to 'ncbi-geneid':  all of input gene IDs are successfully converted.

```r
> head(test.data)
[1] "94"  "230" "241" "301" "302" "664"

> eg2np <- bitr_kegg(test.data, fromType='ncbi-geneid', toType='kegg', organism='hsa') 
Warning message:
In bitr_kegg(test.data, fromType = "ncbi-geneid", toType = "kegg",  :
  100% of input gene IDs are fail to map...

> eg2np 
[1] ncbi-geneid kegg       
<0 行> (或0-长度的row.names)

> eg2np <- bitr_kegg(test.data, fromType='kegg', toType='ncbi-geneid', organism='hsa') 
> eg2np
         kegg ncbi-geneid
1   100049587   100049587
2   100288175   100288175
3       10063       10063
4       10135       10135
5       10148       10148
6       10152       10152
7       10211       10211
8       10261       10261
……
> nrow(eg2np)
[1] 302
```

**Comment by SebastianHesse** [comment]
Yes, seems that the function is broken. I can not get any KEGG ids from the function (it simply copies the input IDs)

**Comment by GuangchuangYu** [comment]
Resolved — the behaviour is correct once the KEGG ID format is the expected one. `bitr_kegg()` matches against the ID column exactly as KEGG returns it, which for `organism = "hsa"` is the **bare** gene number, not the `hsa:`-prefixed form.

Verified both directions on the current release with bare IDs:

```r
bitr_kegg(c("10458","644","840"), fromType = "ncbi-geneid", toType = "kegg", organism = "hsa")
#> rows = 3, 10458 -> 10458, 644 -> 644, 840 -> 840
bitr_kegg(c("10458","644","840"), fromType = "kegg", toType = "ncbi-geneid", organism = "hsa")
#> rows = 3, 10458 -> 10458, ...
```

For human the two ID spaces genuinely coincide (KEGG gene ID = NCBI gene ID), which is why it looks like the input is copied verbatim — it is not, and `ncbi-geneid -> uniprot` on the same IDs returns real UniProt accessions (`10458 -> Q9UQB8`), confirming the conversion table is being used. Passing `"hsa:10458"` is what produced "100% of input gene IDs are fail to map". Closing.
